engine/spec/components/citizens_advice_components/checkbox_group_spec.rb

```diff
@@ -33,6 +33,25 @@
     end
   end
 
+  context "when custom ids are provided" do
+    before do
+      render_inline described_class.new(
+        legend: "Checkbox group field",
+        name: "checkboxes[test][]"
+      ) do |c|
+        c.with_inputs(sample_inputs_with_id)
+      end
+    end
+
+    it "adds the correct name to the input" do
+      expect(page).to have_field "Option 1", name: "checkboxes[test][]"
+    end
+
+    it "adds the correct id to the input" do
+      expect(page).to have_selector "#test-id-first-0"
```

Comment:
I'm not sure if understand why the `-0` is getting added here.

I figure that either:
*  the `id` should be provided at the component level, and the `index` is appended to keep them unique
*  the `id` is provided at the `options` level, so it's up the developer to provide unique ids for each one..?

Reply:
Yeah, after I adjust the `id` to be a parameter on the component, it will be the first case: 
> the id should be provided at the component level, and the index is appended to keep them unique

We've had some issues with id generation in the past and this is when the indexed ids were introduced: https://github.com/citizensadvice/design-system/issues/1729